### ISSUE TITLE
Upgrade for NixOS 20.09

### ISSUE THREAD

**Issue by shadowrylander** [anthe/nixos-surface#6]
Hello! Sorry to bother you again!

When updating to NixOS 20.09 using `RUST_BACKTRACE=1 sudo nixos-rebuild boot --upgrade --show-trace 2>&1 | tee temp`, after fixing the cargo hash mismatches, I get this:
```bash
|Run-time dependency bash-completion found: YES 2.10
   Compiling dirs v1.0.5
   Compiling atty v0.2.13
   Compiling time v0.1.42
error: failed to run custom build command for `libdbus-sys v0.2.1`

Caused by:
  process didn't exit successfully: `/build/surface-dtx-daemon-0.1.4-3/target/release/build/libdbus-sys-855a9e4949284569/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Command { command: "\"pkg-config\" \"--libs\" \"--cflags\" \"dbus-1\" \"dbus-1 >= 1.6\"", cause: Os { code: 2, kind: NotFound, message: "No such file or directory" } }', /build/surface-dtx-daemon-0.1.4-3-vendor.tar.gz/libdbus-sys/build.rs:6:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Again, thank you kindly for all the help!

**Comment by anthe** [comment]
Try adding `pkg-config` to `buildInputs`, the not found error is probably referencing the call to `pkg-config`.

**Comment by shadowrylander** [comment]
Could you possibly give me a code example? NixOS syntax is really confusing to me, and the documentation seems sparse and outdated.

**Comment by anthe** [comment]
To make the change I mentioned change `pkgconfig` to `pkg-config` in `surface-dtx-daemon.nix`. The problem is with `libdbus-sys`, and I don't know enough about the rust build system to say whether making this change for `surface-dtx-daemon` will also add `pkg-config` to the `libdbus-sys` build time PATH, but it's worth a try.

**Comment by shadowrylander** [comment]
Got it; I'll test it out and report back!

**Comment by shadowrylander** [comment]
Well that didn't work. Same error.

**Comment by shadowrylander** [comment]
Yeah; I tried changing all instances of `pkgconfig` to `pkg-config`, and it still doesn't work.

**Comment by shadowrylander** [comment]
Wait... Do you think the rust function is attempting to call the *entire* `pkg-config` command (i.e. `pkg-config --libs --cflags dbus-1 dbus-1 >= 1.6`) as a single command instead of *just* `pkg-config` (i.e. as separate components of a whole command)?

**Comment by shadowrylander** [comment]
Pinging @anthe! Sorry!

**Comment by anthe** [comment]
I am pretty sure rust is calling the entire `pkg-config` command. The NotFound error, I think, is referring to the `pkg-config` command not being found. If `pkg-config` just failed to find dbus then I think you would get some other error.
Have you learned the trick of running `nixos-rebuild -K` to preserve the build directory? Then you can `cd /tmp/nix-build-*`and run `source env-vars` to see what the build environment was like. In this case  I would run `env | grep pkg-config` after running `source env-vars` to see if `pkg-config` is in the `$PATH` or other environment variables.

**Comment by shadowrylander** [comment]
I'll see what I can do; my current build on my other machines isn't quite working either, so I'll have to fix that first! 😅

**Comment by lutzmor** [comment]
Hey :), i found that repo and i'm currently fighting with nixos on my surface pro 7 ;). to solve your problem @shadowrylander you need to do the following in the surface-dtx-daemon.nix

change pkgconfig to pkg-config
buildInputs = [ dbus ];
nativeBuildInputs = [ pgk-config ]

and then your nixos-rebuild switch should come to an successful end :). I will create an pull request if everything is backup and running if wanted :).

Greetings
Moritz

**Comment by shadowrylander** [comment]
Wonderful! However, I'm currently stuck on another major roadblock, if you could possibly help me there: https://www.reddit.com/r/NixOS/comments/jjhw44/review_lib_module_extension/?utm_medium=android_app&utm_source=share

As soon as I get that fixed, I can get back to this!
